ompi/request/request.h

```diff
@@ -448,39 +448,44 @@ static inline bool ompi_request_tag_is_collective(int tag) {
 
 static inline void ompi_request_wait_completion(ompi_request_t *req)
 {
-    if (opal_using_threads () && !REQUEST_COMPLETE(req)) {
-        void *_tmp_ptr;
-        ompi_wait_sync_t sync;
+    if (opal_using_threads ()) {
+        if(!REQUEST_COMPLETE(req)) {
+            void *_tmp_ptr;
+            ompi_wait_sync_t sync;
+
+
 #if OPAL_ENABLE_FT_MPI
-redo:
-        if(OPAL_UNLIKELY( ompi_request_is_failed(req) )) {
-            return;
-        }
+    redo:
```

Comment:
Not your code, not your indentation. Especially as the original one seems pretty fine with me (maybe except the label placement).

Reply:
The indentation needed to be adjusted to account for the additional if at the top, I did so just to keep it consistent. I can move the code within the `OPAL_ENABLE_FT_MPI` blocks back if you want, but in doing so its indentation will be inconsistent with the rest.

Reply:
Why this would not happen if the other thread just completed the request before this thread does the REQUEST_COMPLETE check ? If the code is missing the read barrier in one case then it should be missing it in both isn't it ?